presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/nativechecker/NativePlanCheckerProviderFactory.java

```diff
@@ -0,0 +1,107 @@
+import com.facebook.airlift.json.JsonModule;
+import com.facebook.presto.spi.NodeManager;
+import com.facebook.presto.spi.classloader.ThreadContextClassLoader;
+import com.facebook.presto.spi.plan.PlanCheckerProvider;
+import com.facebook.presto.spi.plan.PlanCheckerProviderContext;
+import com.facebook.presto.spi.plan.PlanCheckerProviderFactory;
+import com.facebook.presto.spi.plan.SimplePlanFragment;
+import com.facebook.presto.spi.plan.SimplePlanFragmentSerde;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Injector;
+import com.google.inject.Scopes;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.util.Map;
+import java.util.Properties;
+
+import static com.facebook.airlift.configuration.ConfigBinder.configBinder;
+import static com.facebook.airlift.json.JsonBinder.jsonBinder;
+import static com.facebook.airlift.json.JsonCodecBinder.jsonCodecBinder;
+import static com.google.common.collect.Maps.fromProperties;
+import static java.util.Objects.requireNonNull;
+
+public class NativePlanCheckerProviderFactory
+        implements PlanCheckerProviderFactory
+{
+    private static final File NATIVE_PLAN_CHECKER_PROVIDER_CONFIG = new File("etc/plan-checker-providers/native-plan-checker.properties");
+    private final ClassLoader classLoader;
+
+    public NativePlanCheckerProviderFactory(ClassLoader classLoader)
+    {
+        this.classLoader = requireNonNull(classLoader, "classLoader is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "native";
+    }
+
+    @Override
+    public PlanCheckerProvider create(Map<String, String> properties, PlanCheckerProviderContext planCheckerProviderContext)
+    {
+        try (ThreadContextClassLoader ignored = new ThreadContextClassLoader(classLoader)) {
+            Bootstrap app = new Bootstrap(
+                    binder -> {
+                        configBinder(binder).bindConfig(NativePlanCheckerConfig.class, NativePlanCheckerConfig.CONFIG_PREFIX);
+                        binder.install(new JsonModule());
+                        binder.bind(NodeManager.class).toInstance(planCheckerProviderContext.getNodeManager());
+                        binder.bind(SimplePlanFragmentSerde.class).toInstance(planCheckerProviderContext.getSimplePlanFragmentSerde());
+                        jsonBinder(binder).addSerializerBinding(SimplePlanFragment.class).to(SimplePlanFragmentSerializer.class).in(Scopes.SINGLETON);
+                        jsonCodecBinder(binder).bindJsonCodec(SimplePlanFragment.class);
+                        binder.bind(PlanCheckerProvider.class).to(NativePlanCheckerProvider.class).in(Scopes.SINGLETON);
+                    });
+
+            Injector injector = app
+                    .noStrictConfig()
+                    .doNotInitializeLogging()
+                    .setRequiredConfigurationProperties(getConfig())
```

Comment:
instead of getting the config yourself, the properties will now be passed to the create() call in the properties field. You can just pass that in directly.